cmd/fabric8-jenkins-proxy/main.go

```diff
@@ -99,11 +99,11 @@ func main() {
 	}
 	mainLogger.WithField("clusters", clusters).Info("Retrieved cluster view")
 
-	start(config, &tenant, wit, idler, store, clusters)
+	start(config, idler, &tenant, wit, store, clusters)
 }
 
-func start(config configuration.Configuration, tenant *clients.Tenant, wit clients.WIT, idler clients.IdlerService, store storage.Store, clusters map[string]string) {
-	proxy, err := proxy.NewProxy(tenant, wit, idler, store, config, clusters)
+func start(config configuration.Configuration, idler clients.IdlerService, tenant clients.TenantService, wit clients.WITService, store storage.Store, clusters map[string]string) {
```

Comment:
@hrishin @kishansagathiya  client.Foo was just fine. How is Suffixing Service to every client adding any more clarity than what it had?

Reply:
Service is suffixed with interface names. 
I went with the flow that we had so far (the package name is `clients`)
I agree that it is an oxymoron. To satisfy the name change you are asking for, we will have to do little rearrangement. I was actually talking about some with @khrm. 

We can have idler, wit, tenant, and auth(why is this kept separate than other clients!) packages inside clients repo. We can get rid of mock as well and move them into respective packages (i.e, mock_idler.go inside `idler` package instead of some `mocks` package)

Reply:
Yea. No need for that auth package. On other hand, I too don't like stuttering. So better to use caps for interface and small letter for concrete implementation. Or define interface where you are going to use it. But latter approach would require separate mock package.